crates/rust-client/src/transactions/mod.rs

```diff
@@ -327,31 +327,27 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store, A: TransactionAuthenticator> Client
         maybe_await!(TransactionResult::new(executed_transaction, screener, future_notes))
     }
 
-    /// Proves the specified transaction witness, and returns a [ProvenTransaction] that can be
-    /// submitted to the node.
-    pub fn prove_transaction(
-        &mut self,
-        executed_transaction: ExecutedTransaction,
-    ) -> Result<ProvenTransaction, ClientError> {
-        let transaction_prover = TransactionProver::new(ProvingOptions::default());
-
-        let proven_transaction = transaction_prover.prove_transaction(executed_transaction)?;
-        Ok(proven_transaction)
-    }
-
-    /// Submits a [ProvenTransaction] to the node, and stores the transaction in
+    /// Proves the specified transaction, submits it to the network, and saves the transaction into
     /// the local database for tracking.
     pub async fn submit_transaction(
         &mut self,
         tx_result: TransactionResult,
-        proven_transaction: ProvenTransaction,
     ) -> Result<(), ClientError> {
+        let transaction_prover = TransactionProver::new(ProvingOptions::default());
+
+        info!("Proving transaction...");
+        let proven_transaction =
+            transaction_prover.prove_transaction(tx_result.executed_transaction().clone())?;
+        info!("Transaction proved.");
```

Comment:
nit: I think I'd write this as 
```suggestion
        info!("Transaction proven.");
```